cranelift-codegen/src/isa/x86/abi.rs

```diff
@@ -689,22 +733,87 @@ fn insert_common_prologue(
             // Also, the size of a return address, implicitly pushed by a x86 `call` instruction,
             // also should be accounted for.
             // TODO: Check if the function body actually contains a `call` instruction.
-            let word_size = isa.pointer_bytes();
             let total_stack_size = (csrs.iter(GPR).len() + 1 + 1) as i64 * word_size as i64;
 
             insert_stack_check(pos, total_stack_size, stack_limit_arg);
         }
     }
 
+    let mut cfa_state = if pos.func.frame_layout.is_some() {
+        Some(CFAState {
+            cf_ptr_reg: RU::rsp as RegUnit,
+            cf_ptr_offset: word_size as isize,
+            current_depth: -(word_size as isize),
+        })
+    } else {
+        None
+    };
+
+    if let Some(ref mut frame_layout) = pos.func.frame_layout {
```

Comment:
Could this `if` be combined with the `if` above? That would result in one fewer `expect` below, and less convoluted code.

Reply:
done!